packages/smart-router/evm/v3-router/functions/getPairCombinations.ts

```diff
@@ -29,21 +27,15 @@ const getToken = memoize(
   (chainId, address) => `${chainId}_${address}`,
 )
 
-// TODO: move to gauges
-const getGaugesByChain = memoize(
-  (chainId?: ChainId): GaugeConfig[] => allGauges.filter((gauge) => gauge.chainId === chainId),
-  (chainId) => chainId,
-)
-
 function isTokenInCommonBases(token?: Token) {
   return Boolean(token && BASES_TO_CHECK_TRADES_AGAINST[token.chainId as ChainId]?.find((t) => t.equals(token)))
 }
 
 const getTokenBasesFromGauges = memoize(
-  (currency?: Currency): Token[] => {
+  async (currency?: Currency): Promise<Token[]> => {
     const chainId: ChainId | undefined = currency?.chainId
     const address = currency?.wrapped.address
-    const gauges = getGaugesByChain(currency?.chainId)
+    const gauges = await getGaugesByChain(chainId)
```

Comment:
This is something we can cache and don't need to fetch every time.

Also we might need some retry strategy to make sure all data is fetched. Meanwhile we also need to make sure that even if the fetch failed, rest logic still works as it's not the main flow

Reply:
we need to make sure:
1. Even if we failed to get gauges, we can still proceed with swap
2. We can't let gauge config become the bottleneck which slows down the whole swap. So it's better to add a timeout